osc-server/src/main/java/org/osc/core/broker/service/validator/DeleteDistributedApplianceRequestValidator.java

```diff
@@ -53,13 +59,46 @@ public DistributedAppliance validateAndLoad(BaseDeleteRequest request) throws Ex
                             da.getId()));
         }
 
+        validateDAReferenceToSfcAndSecurityGroup(em, da);
+
         if (!da.getMarkedForDeletion() && request.isForceDelete()) {
             throw new VmidcBrokerValidationException(
                     "Distributed Appilance with ID "
                             + request.getId()
                             + " is not marked for deletion and force delete operation is applicable only for entries marked for deletion.");
         }
-
         return da;
     }
+
+    private void validateDAReferenceToSfcAndSecurityGroup(EntityManager em, DistributedAppliance da) throws Exception {
+        for (VirtualSystem vs : da.getVirtualSystems()) {
+            try {
+                validateVSReferenceToSfcAndSecurityGroup(em, vs);
+            } catch (VmidcBrokerValidationException ex) {
+                throw new VmidcBrokerValidationException(String.format("Cannot delete the distributed appliance '%s' reason:",da.getName())+" "+ex.getMessage());
```

Comment:
`"Cannot delete the distributed appliance '%s' reason: ", da.getName()) + ex.getMessage()) ` Please attempt to the spaces and details.